File: algorithms/interview-preparation-kit-1/js/code-response/arrayStats/arrayStats.js
function stats(arr) {
  const stat = [0, 0, 0];
  arr.forEach(e => {
    if (e < 0) stat[0] += 1;
    else if (e === 0) stat[1] += 1;
    else stat[2] += 1;
  });

  return stat.map(e => Number(e / arr.length).toFixed(6));
}

export { stats };

File: algorithms/interview-preparation-kit-1/ruby/code-response/arrayStats/array_stats.rb
class ArrayStats
    def ArrayStats.stats(arr)
        status = [0, 0, 0]

        arr.each do |e|
            if e < 0
                status[0] += 1
            elsif e === 0
                status[1] += 1
            else
                status[2] += 1
            end
        end

        return status.map do |stat|
            '%.6f' % (stat.to_f / arr.size)
        end
    end

end